R/getPrevalence.R

```diff
@@ -258,8 +290,17 @@ setGeneric("getPrevalentTaxa", signature = "x",
     } else {
         taxa <- pr > prevalence
     }
-
-    taxa <- names(which(taxa))
+    
+    # Gets indices of most prevalent taxa
+    indices  <- which(taxa)
+    # If vector contains names
+    if( !is.null(names(indices)) ){
+        # Gets the names
+        taxa <- names(indices)
+    } else {
+        # Otherwise indices are returned
+        taxa <- indices
+    }
     taxa
```

Comment:
Regarding the subsetting. I'm still thinking what kind of value check of `rownames(x)` in `subsetByPrevalentTaxa` brings and also how it should be then constructed. 

This part is from `getPrevalentTaxa`. So, should this part be constructed differently? `getPrevalentTaxa` and `getRareTaxa` actually return numeric vector, when `rownames` is `NULL`. 

```
> x <- GlobalPatterns
> y <- x
> rownames(y) <- NULL
> is.numeric(getPrevalentTaxa(x))
[1] FALSE
> is.numeric(getPrevalentTaxa(y))
[1] TRUE
```

Reply:
That is certainly a bit weird, since both results are generated from `names(which())` once with an unnamed input and once with a named input. I first thought this was due to changes from R 4.0 to R 4.1, but apparently it is not. I have to check with release though.

Reply:
What is the result of `getPrevalentTaxa(y)`? It should be `NULL` since the names are not set.

Reply:
This part does it (rows 294-304)

```
    # Gets indices of most prevalent taxa
    indices  <- which(taxa)
    # If vector contains names
    if( !is.null(names(indices)) ){
        # Gets the names
        taxa <- names(indices)
    } else {
        # Otherwise indices are returned
        taxa <- indices
    }
    taxa
```

Reply:
`getPrevalentTaxa(y)` return currently indices / numeric vector, but I can change that

Reply:
Yeah I didn't pull the commits properly. Let me check again

Reply:
So I think this needs some in depth refactoring. WIth all the names indices and such, it has gotten quite confusing. Give me a few minutes to come up with something

Reply:
So this was a bit tricky, but I think the clear seperation for returning indices and names is helpful. Have a look at the changes and let me know, what you think about the changes.

Reply:
Yep, now it looks very good!